pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go

```diff
@@ -0,0 +1,358 @@
+		tkgCtlClient  tkgctl.TKGClient
+		logsDir       string
+		clusterName   string
+		namespace     string
+		mcProxy       *framework.ClusterProxy
+		mcContextName string
+		timeout       time.Duration
+	)
+	const (
+		waitTimeout     = time.Minute * 15
+		pollingInterval = time.Second * 10
+	)
+
+	BeforeEach(func() { //nolint:dupl
+		namespace = constants.DefaultNamespace
+		input = inputGetter()
+		if input.Namespace != "" {
+			namespace = input.Namespace
+		}
+		timeout, err = time.ParseDuration(input.E2EConfig.DefaultTimeout)
+		rand.Seed(time.Now().UnixNano())
+		mcClusterName := input.E2EConfig.ManagementClusterName
+		mcContextName = mcClusterName + "-admin@" + mcClusterName
+		mcProxy = framework.NewClusterProxy(mcClusterName, "", mcContextName)
+
+		logsDir = filepath.Join(input.ArtifactsFolder, "logs")
+
+		rand.Seed(time.Now().UnixNano())
+		clusterName = input.E2EConfig.ClusterPrefix + "wc-" + util.RandomString(4) // nolint:gomnd
+
+		tkgCtlClient, err = tkgctl.New(tkgctl.Options{
+			ConfigDir: input.E2EConfig.TkgConfigDir,
+			LogOptions: tkgctl.LoggingOptions{
+				File:      filepath.Join(logsDir, clusterName+".log"),
+				Verbosity: input.E2EConfig.TkgCliLogLevel,
+			},
+		})
+
+		Expect(err).To(BeNil())
+	})
+
+	It("Should create workload cluster with default kubernetes version and verify infra machine images are resolved correctly", func() {
+		By(fmt.Sprintf("Creating a workload cluster %q", clusterName))
+
+		options := framework.CreateClusterOptions{
+			ClusterName:  clusterName,
+			Namespace:    namespace,
+			Plan:         "dev",
+			CniType:      input.Cni,
+			OtherConfigs: input.OtherConfigs,
+		}
+		if input.Plan != "" {
+			options.Plan = input.Plan
+		}
+
+		if input.E2EConfig.InfrastructureName == "vsphere" {
+			if input.Cni == "antrea" {
+				if clusterIP, ok := os.LookupEnv("CLUSTER_ENDPOINT_ANTREA"); ok {
+					options.VsphereControlPlaneEndpoint = clusterIP
+				}
+			}
+			if input.Cni == "calico" {
+				if clusterIP, ok := os.LookupEnv("CLUSTER_ENDPOINT_CALICO"); ok {
+					options.VsphereControlPlaneEndpoint = clusterIP
+				}
+			}
+		}
+
+		clusterConfigFile, err := framework.GetTempClusterConfigFile(input.E2EConfig.TkgClusterConfigPath, &options)
+		Expect(err).To(BeNil())
+
+		defer os.Remove(clusterConfigFile)
+
+		err = tkgCtlClient.CreateCluster(tkgctl.CreateClusterOptions{
+			ClusterConfigFile: clusterConfigFile,
+			Edition:           "tkg",
+			Namespace:         namespace,
+		})
+		Expect(err).To(BeNil())
+
+		By(fmt.Sprintf("Validating the TKR data after cluster %q is created", clusterName))
+		verifyTKRData(context, mcProxy, options.ClusterName, options.Namespace)
+
+		By(fmt.Sprintf("Deleting workload cluster %q", clusterName))
+		err = tkgCtlClient.DeleteCluster(tkgctl.DeleteClustersOptions{
+			ClusterName: clusterName,
+			Namespace:   namespace,
+			SkipPrompt:  true,
+		})
+		Expect(err).To(BeNil())
+
+		By("Test successful !")
```

Comment:
Cluster tear-down is identical between the 2 test cases, as well. So, it could be factored out and put into an `AfterEach()`